### PR TITLE
DVC pipeline itterates over pgdata splits instead of pgdatasets

### LINKED ISSUE #114: running pg-benchmark does not take into account dataset splits.
### Summary

I realise we don't have the splitting functionality in the dataset yet, but should not forget to implement this in the benchmark when possible. 

Current examples do not take a split dataset, they take the full dataset and set this to the supervise/zero-shot benchmark. 

### Motivation

To avoid confusion we should only utilize:

- train, validation, test datasets in supervised.
- test dataset in zero-shot. 

### Proposed solution

Depends on execution of https://github.com/ProteinGym/proteingym-base/issues/294

### PR DESCRIPTION
**Pull request by karel-w**
# Changes

Resolves #114 

Its finally done! Tested on dms subs & indels for zero-shot and supervised. Not yet tested on all datasets, thats the next step. 

- removed old dvc cache (no sense storing cache for now)
- metrics.csv / datasets.json / models.json are with settings from last run experiments. We should commit a full dataset list with all datasets for release.
- We itterate over each fold within the dataset now, so e.g. when we select fold 0, fold 0 is the test set.
- Changed pls model and esm model to use splits

# Checklist

- [ ] I broke the PR down so that it contains a reasonable amount of changes for an effective review
- [x] I performed a self-review of my code. Amongst other things, I have commented my code in hard-to-understand areas.
- [x] I made corresponding changes to the documentation
- [x] I added tests that prove my fix is effective or that my feature works
- [x] I accounted for dependent changes to be merged and published in downstream modules

### CODE REVIEW

File: scripts/utils.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                    variance = sum((x - mean) ** 2 for x in fold_values) / len(fold_values)
+                    stdev = math.sqrt(variance)
+                else:
+                    stdev = 0.0
+                rows.append(f"{game},{model},{dataset},{target},{mean},{stdev}")
+    
+    output_path.write_text('\n'.join(rows) + '\n')
+
+
+def main():
```
Comment (hredestig) [inline scripts/utils.py]: we use typer elsewhere and it's so nice, maybe here as well?

File: models/esm/src/proteingym/models/esm/preprocess.py [hunk 2]
```diff
@@ -1,5 +1,95 @@
...
+            return f"{wt_aa}{i+1}{mt_aa}"
+    return ""
+
+
+def load_x_and_y(
+    subset: Subsets,
+    split: str,
+    test_fold: int,
+    target: str,
+    wt_seq: str = None,
```
Comment (hredestig) [inline models/esm/src/proteingym/models/esm/preprocess.py]: `str | None = None`

File: models/esm/src/proteingym/models/esm/utils.py [hunk 3]
```diff
@@ -133,6 +97,18 @@ def compute_pppl(
...
 
     return sum(log_probs)
+
+
+def compute_pppl_from_mutation(
+    row: str, sequence: str, model: object, alphabet: object, offset_idx: int
+) -> float:
+    """Compute pseudo-perplexity for a mutation string by applying it to the sequence."""
+    wt, idx, mt = row[0], int(row[1:-1]) - offset_idx, row[-1]
+    assert sequence[idx] == wt, (
```
Comment (hredestig) [inline models/esm/src/proteingym/models/esm/utils.py]: ValueError - asserts have different purpose https://nocomplexity.substack.com/p/stop-using-assert-in-your-python

File: models/esm/src/proteingym/models/esm/preprocess.py [hunk 4]
```diff
@@ -1,5 +1,95 @@
...
+) -> tuple[list[Any], list[Any], list[Any], list[Any], bool]:
+    """Load train/test splits for k-fold cross-validation.
+    
+    Args:
+        subset: a split dataset in the Subsets format
+        split: name of the split
+        test_fold: Which kfold split to take as test set
+        target: name of the target we are classifying
+        wt_seq: Wild-type sequence for converting mutant sequences to mutation strings
+        return_sequences: If True, return full sequences; if False, return mutation strings
```
Comment (hredestig) [inline models/esm/src/proteingym/models/esm/preprocess.py]: why mutation strings?

File: scripts/utils.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
+import argparse
+import json
+import math
+from pathlib import Path
+import numpy as np
+import polars as pl
+
+
+def aggregate_metrics(metric_dir: Path, dataset_name: str, model_name: str, output_path: Path, prediction_dir: Path = None):
```
Comment (hredestig) [inline scripts/utils.py]: too complex and would look a lot leaner if the metadata was in the json file instead of encoded in the path

**Review by hredestig — CHANGES_REQUESTED** [review]

File: models/esm/src/proteingym/models/esm/preprocess.py [hunk 6]
```diff
@@ -1,5 +1,95 @@
...
+
+import polars as pl
 import torch
 from esm.data import Alphabet
+from proteingym.base import Subsets
+
+logger = logging.getLogger(__name__)
+
+
+def has_indels(mutant_seq: str, wt_seq: str) -> bool:
```
[the remark this answers is not in the document]
Comment (karel-w) [reply models/esm/src/proteingym/models/esm/preprocess.py]: Should indels be a flag on the dataset maybe? This is definitely a very rudimentary way (and incorrect) way to check for indels. But properly checking for indels is not that straightforward, e.g. 1 insertion 1 deletion can shift the sequence. 

Atleast for the ESM model we need to know if we have indels or not for deciding which scoring strategy.

File: scripts/utils.py [hunk 7]
```diff
@@ -0,0 +1,144 @@
+import json
```
Comment (hredestig) [inline scripts/utils.py]: this all would benefit from pydantic

**Review by hredestig — APPROVED** [review]
Only remarks for future changes
